analysis/quantifier-comparisons/10-miQC-filtering.Rmd

```diff
@@ -0,0 +1,272 @@
+
+```{r}
+# function to test miQC modeling with different parameters on sce objects and show plots
+model_sce <- function(sce, title, model_type = "linear", posterior_cutoff = 0.75){
+  
+  # plot distributions of total counts/cell, genes detected/cell and mito content
+  coldata_df <- data.frame(colData(sce))
+  manual_filter_plot <- ggplot(coldata_df, aes(x = sum, y = detected, color = subsets_mito_percent)) + 
+    geom_point(alpha = 0.5) +
+    scale_color_viridis_c() + 
+    labs(x = "Total Count",
+         y = "Number of Genes Expressed",
+         color = "Mitochondrial\nFraction") + 
+    geom_vline(xintercept = 500) +
+    geom_hline(yintercept = 1000) +
+    ggtitle(title) +
+    theme_classic()
+  
+  # create model
+  sce_model <- miQC::mixtureModel(sce, model_type)
+  
+  # plot cells colored by probability of cell being compromised or not
+  model_plot <- miQC::plotModel(sce, sce_model) +
+    theme_classic()
+  
+  # plot cells that would be filtered using model vs cells that would be filtered using manual QC cutoffs
+  miQC_filter_plot <- miQC::plotFiltering(sce, sce_model, posterior_cutoff) + 
+    geom_hline(yintercept = 20) +
+    geom_vline(xintercept = 500) + 
+    theme_classic()
+  
+  # arrange plots
+  grid.arrange(manual_filter_plot, model_plot, miQC_filter_plot, nrow = 2)
+}
+```
+
+
+```{r fig.height=5, fig.width=10}
+purrr::map2(cr_like_sce, names(cr_like_sce), model_sce)
+```
+Using the suggested parameters from miQC we see that it does a fairly good job of keeping cells with low mito content and high number of genes detected for the single-cell samples. 
+For the single-nuclei samples however, it looks like the model is throwing away a lot more cells than we may want to probably due to the low range in mitochondrial percentages to begin with. 
+Let's see if we alter some of the parameters used, like the cutoff for filtering, or the model used to compute the probability of compromised cells, and if that impacts the output. 
+
+```{r}
+# change the posterior cutoff for miQC to keep throw away cells with > 0.9 probability of being compromised
+purrr::map2(cr_like_sce, names(cr_like_sce), model_sce, posterior_cutoff = 0.9)
+```
+
+```{r}
+# test spline mixture model rather than linear 
+purrr::map2(cr_like_sce, names(cr_like_sce), model_sce, model_type = "spline")
+```
+
+```{r}
+# test polynomial mixture model rather than linear
+purrr::map2(cr_like_sce, names(cr_like_sce), model_sce, model_type = "polynomial")
+```
+
+It looks like the change in model or different posterior cutoff doesn't have a drastic change in the cells that get included or excluded for this set of single-cell or single-nuclei samples. 
+
+## Filter Cells
+
+Now we can compare filtering using miQC vs. using the pre-determined cutoffs to see how the different filtering effects the overall number of cells that are removed. 
```

Comment:
I think it is worth mentioning here what these cutoffs are going to be. I can see from the code that you used 20% mito, but a reader might not see that easily, and the genes detected & count thresholds are not obvious either. 

My understanding of miQC is that it will keep low genes detected and counts, as the model doesn't really look for that, focusing on mitchondrial content (mediated by gene #). I would think we might want to use those even with miQC, so I feel like including them here might make for a bit of a biased comparison.  

Also, the total count threshold should largely be handled largely by emptyDrops: I don't think we want to use that as a hard cutoff after using emptyDrops specifically to avoid that hard cutoff!